Source/WarpX.cpp

```diff
@@ -993,25 +989,23 @@ WarpX::ReadParameters ()
         pp_lasers.queryarr("names", lasers_names);
 
         std::vector<std::string> sort_intervals_string_vec = {"-1"};
+        int particle_shape;
         if (!species_names.empty() || !lasers_names.empty()) {
-            int particle_shape;
-            if (queryWithParser(pp_algo, "particle_shape", particle_shape) == false)
-            {
-                amrex::Abort("\nalgo.particle_shape must be set in the input file:"
-                             "\nplease set algo.particle_shape to 1, 2, or 3");
+            if (queryWithParser(pp_algo, "particle_shape", particle_shape)){
+
+                WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+                    (particle_shape >= 1) && (particle_shape <=3),
+                    "algo.particle_shape can be only 1, 2, or 3"
+                );
+
+                nox = particle_shape;
```

Comment:
At some point we should rebase out these three to a single control variable, since we do not support non-isotropic shapes anymore.

Reply:
I agree!